compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1181,13 +1181,37 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] w
    *  for equality would give the wrong result, so we should not use the sets
    *  for comparisons.
    */
-  def canCompare(atoms: Set[Type]): Boolean =
-    ctx.phase.isTyper || {
+  def compareAtoms(tp1: Type, tp2: Type): Option[Boolean] =
+    def canCompare(ts: Set[Type]) = ctx.phase.isTyper || {
       val hasSkolems = new ExistsAccumulator(_.isInstanceOf[SkolemType]) {
         override def stopAtStatic = true
       }
-      !atoms.exists(hasSkolems(false, _))
+      !ts.exists(hasSkolems(false, _))
     }
+    def verified(result: Boolean): Boolean =
+      if Config.checkAtomsComparisons && false then
```

Comment:
Since `checkAtomsComparisons is a final val this is equivalent to:
```suggestion
      if Config.checkAtomsComparisons then
```